packages/react-reconciler/src/ReactFiberHooks.js

```diff
@@ -3261,7 +3261,7 @@ export function startHostTransition<F>(
 
 function ensureFormComponentIsStateful(formFiber: Fiber) {
   const existingStateHook: Hook | null = formFiber.memoizedState;
-  if (existingStateHook !== null) {
+  if (existingStateHook !== null && existingStateHook.next !== null) {
```

Comment:
I'm not sure if this is the best way to do this, this might just make it so this if statement is never true

Reply:
Fixed in #33055, removing this change